src/microbe_stage/CompoundBag.cs

```diff
@@ -38,13 +33,30 @@ public CompoundBag(float capacity)
     /// <summary>
     ///   Gets the capacity for a given compound
     /// </summary>
-    /// <returns>Returns <see cref="Capacity"/> if the compound is useful, otherwise 0</returns>
+    /// <returns>Returns CompoundCapacities[compound] if the compound is useful, otherwise 0</returns>
     public float GetCapacityForCompound(Compound compound)
     {
-        if (IsUseful(compound))
-            return Capacity;
+        if (!IsUseful(compound))
+            return 0;
 
-        return 0;
+        if (compoundCapacities != null && compoundCapacities.TryGetValue(compound, out var capacity))
+            return capacity;
+
+        return nominalCapacity;
+    }
+
+    /// <summary>
+    ///   Sets the capacity for all compounds
```

Comment:
I think this comment needs to elaborate that this is the capacity for all compounds that don't have a specific capacity set.